### ISSUE TITLE
Redefinition of the studies dataset

### ISSUE THREAD

**Issue by ireneisdoomed** [opentargets/issues#2827]
This issue captures the changes needed to consolidate the studies schema. Apart from reviewing the necessity of the current fields, the main effort is the addition of QTL studies as part of the study index.

## Background

Many of us are currently working on pipelines which require having a consolidated studies dataset table. I think it is a good idea coming up with some common ground of what this index might look like so that the new data that we produce is as close to a real scenario as possible (similar effort as #2826).

QTL studies are currently not part of the studies. We want to treat them similarly as GWASs, with the particularity that we have to think about what defines a QTL study.

## Tasks

- [ ] Agree on a definition of study for QTLs to come up with an ID. The proposal is using the source of the study, the measured trait and the biofeature (tissue, e.g.) where it is measured to create a hash that will uniquely identify it.
- [ ] Review the existing fields
- [ ] Get the necessary metadata for QTLs 

## Acceptance tests

TBD

**Comment by ireneisdoomed** [comment]
The revision of this schema is captured in the spreadsheet (study_changes_v5) https://docs.google.com/spreadsheets/d/17Rjyv6YB5iXk0831gGvUNLgTaM2_jjTmp517Xg7iWSU/edit#gid=706621111

The data is stored here "gs://genetics_etl_python_playground/input/study_index"

As explained above, and as a temporary solution (or not), ids for QTL studies are generated by hashing `source`, `traitFromSource`, `biofeature` and the qtl `type`.

This is the resulting schema for which I have also taken into account the model in the newly produced GWASCatalog study table by @DSuveges.
```
root
 |-- id: string (nullable = true)
 |-- ancestryInitial: array (nullable = true)
 |    |-- element: string (containsNull = true)
 |-- ancestryReplication: array (nullable = true)
 |    |-- element: string (containsNull = true)
 |-- nCases: long (nullable = true)
 |-- nSamples: long (nullable = true)
 |-- nControls: long (nullable = true)
 |-- pubmedId: string (nullable = true)
 |-- publicationFirstAuthor: string (nullable = true)
 |-- publicationDate: string (nullable = true)
 |-- publicationJournal: string (nullable = true)
 |-- publicationTitle: string (nullable = true)
 |-- traitFromSource: string (nullable = true)
 |-- hasSummaryStats: boolean (nullable = true)
 |-- source: string (nullable = true)
 |-- traitFromSourceMappedIds: array (nullable = true)
 |    |-- element: string (containsNull = true)
 |-- traitCategory: string (nullable = true)
 |-- type: string (nullable = true)
 |-- backgroundTraitFromSourceMappedIds: string (nullable = true)
 |-- initialSampleSize: string (nullable = true)
 |-- summaryStatsLocation: string (nullable = true)
 |-- biofeature: string (nullable = true)
 |-- biofeatureId: string (nullable = true)
```
And the code to produce it is collected here: https://gist.github.com/ireneisdoomed/da6e7ba344a50651ebbe860224907243

**Comment by ireneisdoomed** [comment]
After a discussion with @d0choa, we have decided that we want to capture in this table what is the mapped gene that has been derived from the measured phenotype. This information is currently available in `gs://genetics_etl_python_playground/input/phenotype_id_gene_luts` and it assigns a gene for all QTL studies that do not have an ENGSID in the reported trait already.
This dependency is part of (at least) the colocalisation and the L2G pipeline, so adding the metadata in the study table removes this dependency from those pipelines.

Both the gist and the data above have been updated to reflect this.
